integrations/event-handler/session_events_job.go

```diff
@@ -147,6 +119,119 @@ func (j *SessionEventsJob) run(ctx context.Context) error {
 	}
 }
 
+func (j *SessionEventsJob) processSession(ctx context.Context, s session, processingAttempt int) error {
+	const (
+		// maxNumberOfProcessingAttempts is the number of times a non-existent
+		// session recording will be processed before assuming the recording
+		// is lost forever.
+		maxNumberOfProcessingAttempts = 2
+		// eventTimeCutOff is the point of time in the past at which a missing
+		// session recording is assumed to be lost forever.
+		eventTimeCutOff = -48 * time.Hour
+	)
+
+	log := logger.Get(ctx).WithField("id", s.ID).WithField("index", s.Index)
+	backoff := backoff.NewDecorr(sessionBackoffBase, sessionBackoffMax, clockwork.NewRealClock())
+	attempt := sessionBackoffNumTries
+	retry, err := j.consumeSession(ctx, s)
+
+	for {
```

Comment:
```suggestion

	for {
		retry, err := j.consumeSession(ctx, s)
```

should this be moved into the for loop? 
otherwise retry and err will be static and we will keep ranging the same results until the break condition is triggered